sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CollectionExpressionsSuite.scala

```diff
@@ -1836,4 +1836,16 @@ class CollectionExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper
     checkEvaluation(ArrayIntersect(empty, oneNull), Seq.empty)
     checkEvaluation(ArrayIntersect(oneNull, empty), Seq.empty)
   }
+
+  test("SPARK-31982: sequence doesn't handle date increments that cross DST") {
+    Array("America/Chicago", "GMT", "Asia/Shanghai").foreach(tz => {
+      checkEvaluation(Sequence(
+        Cast(Literal("2011-03-01"), DateType),
+        Cast(Literal("2011-04-01"), DateType),
+        Option(Literal(stringToInterval("interval 1 month"))),
+        Option(tz)),
+        Seq(
+          Date.valueOf("2011-03-01"), Date.valueOf("2011-04-01")))
```

Comment:
The guys are still in America/Los_Angeles, right?

Reply:
Yes, `America/Los_Angeles` can pass the test.

Reply:
Sure, the result could be `tz` independently.

Reply:
Please, wrap the code by `withDefaultTimeZone` otherwise your expected dates are wrong.